### PR TITLE
Update Juice-Shop Helm chart to use modern Ingress resource

### PR DESCRIPTION
**Pull request by maze88**
## Description

The Juice-Shop demo-target application can be installed to Kubernetes using a provided Helm chart which also allows exposing its service online via an Ingress resource. This Ingress resource's template (in the Helm chart) was using [an out-dated API version](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122).

This pull request contributes an up to date version of the Ingress resource's template to use the standard `networking.k8s.io/v1` API version, instead of the old `networking.k8s.io/v1beta1` or `extensions/v1beta1` API versions (conditionally).

## Testing
In order to test one can install with Helm to a Kubernetes cluster, or render the Kubernetes manifests with `helm template` and compare the YAML to the modern structure here.

<details>
<summary>Click to expand <em>my</em> test details...</summary>

With the values file:
```yaml
# test-values.yaml
ingress:
  enabled: true
  className: nginx
  pathtype: ImplementationSpecific
  annotations: {}
  hosts:
  - host: juice.maze.omg.lol
    paths:
    - path: /
  tls:
  - hosts:
    - juice.maze.omg.lol
    secretName: juice-shop-tls
```
deploy with:
```bash
helm install my-juice . -f test-values.yaml
```
or run:
```bash
helm template my-juice . -f test-values.yaml | grep -A 26 Ingress
```
which yields:
```yaml
kind: Ingress
metadata:
  name: my-juice-juice-shop
  labels:
    helm.sh/chart: juice-shop-v3.1.0-alpha3
    app.kubernetes.io/name: juice-shop
    app.kubernetes.io/instance: my-juice
    app.kubernetes.io/version: "v13.0.3"
    app.kubernetes.io/managed-by: Helm
spec:
  ingressClassName: "nginx"
  tls: 
  - hosts:
    - juice.maze.omg.lol
    secretName: juice-shop-tls
  rules:
  - host: juice.maze.omg.lol
    http:
      paths:
      - path: /
        pathType: ImplementationSpecific
        backend:
          service:
            name: my-juice-juice-shop
            port:
              number: 3000
---

```
</details>

### CODE REVIEW

**Comment by Zero3141** [comment]
Hi maze88,

thank you for your contribution, looks good for us. Please sign your commits according to our [CONTRIBUTING.md](https://github.com/secureCodeBox/secureCodeBox/blob/main/CONTRIBUTING.md) and add yourself to the [CONTRIBUTORS.md](https://github.com/secureCodeBox/secureCodeBox/blob/main/CONTRIBUTORS.md) file, then we can merge. :)

TLDR for signing commits:
1. Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
2. In your local branch, run: git rebase HEAD~5 --signoff
3. Force push your changes to overwrite the branch: git push --force-with-lease origin main

**Comment by maze88** [comment]
Due to merges of the upstream to my fork the history and signoffs became botched.
I shall re-fork and recreate the PR; standby.
